src/api/app/models/project.rb

```diff
@@ -869,6 +873,10 @@ def expand_all_repositories
     all_repositories.uniq
   end
 
+  def expand_linking_to
+    expand_all_projects(allow_remote_projects: false).map(&:id)
```

Comment:
Yeah I'm contemplating of re-factoring expand_all_projects to only return ids and then instantiate the objects only when needed. Maybe next time...

![giphy](https://user-images.githubusercontent.com/514785/28877323-5359efa2-779c-11e7-9b53-7a7d47ac60be.gif)